### PR TITLE
[3.9] Filesystem namespace

### PR DESCRIPTION
**Pull request by fastslack**
## Summary of Changes

Moved all libraries names to use namespace

## Testing Instructions

Use as usual, all should work fine.

## Expected result

No errors

### CODE REVIEW

File: libraries/src/Filesystem/Streams/String.php [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+ * Joomla! Content Management System
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license     GNU General Public License version 2 or later; see LICENSE
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
-defined('JPATH_PLATFORM') or die;
+namespace Joomla\CMS\Filesystem\Streams;
```
Comment (laoneo) [inline libraries/src/Filesystem/Streams/String.php]: Namespace declaration must be on top of the file.

File: libraries/src/Filesystem/File.php [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
+use Joomla\CMS\Factory;
+use Joomla\CMS\Log\Log;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Filesystem\Path;
+use Joomla\CMS\Filesystem\Wrapper\PathWrapper;
+use Joomla\CMS\Filesystem\Wrapper\FolderWrapper;
+use Joomla\CMS\Client\ClientHelper;
+use Joomla\CMS\Client\FtpClient;
+
 defined('JPATH_PLATFORM') or die;
```
Comment (laoneo) [inline libraries/src/Filesystem/File.php]: Put that after the namespace declaration.

File: libraries/src/Filesystem/File.php [hunk 3]
```diff
@@ -1,20 +1,31 @@
...
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license     GNU General Public License version 2 or later; see LICENSE
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+namespace Joomla\CMS\Filesystem;
+
+
```
Comment (laoneo) [inline libraries/src/Filesystem/File.php]: One blank line too much.

File: libraries/src/Filesystem/Folder.php [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Log\Log;
+use Joomla\CMS\Filesystem\Wrapper\PathWrapper;
+use Joomla\CMS\Filesystem\Wrapper\FileWrapper;
+use Joomla\CMS\Client\ClientHelper;
+use Joomla\CMS\Client\FtpClient;
+use Joomla\CMS\Language\Text;
+
 defined('JPATH_PLATFORM') or die;
```
Comment (laoneo) [inline libraries/src/Filesystem/Folder.php]: Move after namespace declaration.

File: libraries/src/Filesystem/Patcher.php [hunk 5]
```diff
@@ -1,24 +1,25 @@
...
  */
 
-defined('JPATH_PLATFORM') or die;
+namespace Joomla\CMS\Filesystem;
 
-jimport('joomla.filesystem.file');
+use Joomla\CMS\Log\Log;
+use Joomla\CMS\Language\Text;
+
+defined('JPATH_PLATFORM') or die;
```
Comment (laoneo) [inline libraries/src/Filesystem/Patcher.php]: Move after namespace declaration.

File: libraries/src/Filesystem/Path.php [hunk 6]
```diff
@@ -1,18 +1,23 @@
...
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+namespace Joomla\CMS\Filesystem;
+
+use Joomla\CMS\Filesystem\Wrapper\PathWrapper;
+use Joomla\CMS\Filesystem\Wrapper\FileWrapper;
+use Joomla\CMS\Crypt\Crypt;
+
 defined('JPATH_PLATFORM') or die;
```
Comment (laoneo) [inline libraries/src/Filesystem/Path.php]: Move after namespace declaration.

File: libraries/src/Filesystem/Stream.php [hunk 7]
```diff
@@ -1,19 +1,24 @@
...
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+namespace Joomla\CMS\Filesystem;
+
+use Joomla\CMS\Object\CMSObject;
+use Joomla\Filesystem\FilesystemHelper;
+use Joomla\Language\Text;
+
 defined('JPATH_PLATFORM') or die;
```
Comment (laoneo) [inline libraries/src/Filesystem/Stream.php]: Move after namespace declaration.

File: tests/unit/suites/libraries/joomla/filesystem/JFilesystemHelperTest.php [hunk 8]
```diff
@@ -61,7 +61,7 @@ public function testGetJStreams()
 		$streams = JFilesystemHelper::getJStreams();
 
 		$this->assertEquals(
-			array('string'),
+			array('String'),
```
Comment (laoneo) [inline tests/unit/suites/libraries/joomla/filesystem/JFilesystemHelperTest.php]: Should this not be lower case?

File: tests/unit/suites/libraries/joomla/filesystem/JFilesystemHelperTest.php [hunk 9]
```diff
@@ -76,7 +76,7 @@ public function testGetJStreams()
 	public function testIsJoomlaStream()
 	{
 		$this->assertTrue(
-			JFilesystemHelper::isJoomlaStream('string')
+			JFilesystemHelper::isJoomlaStream('String')
```
Comment (laoneo) [inline tests/unit/suites/libraries/joomla/filesystem/JFilesystemHelperTest.php]: Should this not be lower case?

**Review by laoneo — CHANGES_REQUESTED** [review]

**Comment by laoneo** [comment]
There are still unit tests failing. Can you fix them please.

**Comment by fastslack** [comment]
@laoneo @mbabker Im still having the same issue. Test fail is

```
1) JFilesystemHelperTest::testGetJStreams
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'string'
+    0 => 'String'
 )
C:\projects\joomla-cms\tests\unit\suites\libraries\joomla\filesystem\JFilesystemHelperTest.php:65
```

But if i change that file, i still having testing issues. Any idea?

**Comment by laoneo** [comment]
There are some other errors, please apply the patch on your branch. Then the tests should pass and we can get that in.

```
Index: libraries/src/Filesystem/Streams/String.php
===================================================================
--- libraries/src/Filesystem/Streams/String.php	(date 1515624926000)
+++ libraries/src/Filesystem/Streams/StreamString.php	(date 1517480939000)
@@ -290,4 +290,4 @@
 	}
 }
 
-stream_wrapper_register('string', 'StreamString') or die('StreamString Wrapper Registration Failed');
+stream_wrapper_register('string', '\\Joomla\\CMS\\Filesystem\\Streams\\StreamString') or die('StreamString Wrapper Registration Failed');
Index: libraries/src/Filesystem/Support/Stringcontroller.php
===================================================================
--- libraries/src/Filesystem/Support/Stringcontroller.php	(date 1515624926000)
+++ libraries/src/Filesystem/Support/StringController.php	(date 1517480969000)
@@ -15,7 +15,7 @@
  *
  * @since  11.1
  */
-class Stringcontroller
+class StringController
 {
 	/**
 	 * Defines a variable as an array
Index: libraries/src/Filesystem/FilesystemHelper.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- libraries/src/Filesystem/FilesystemHelper.php	(date 1515624926000)
+++ libraries/src/Filesystem/FilesystemHelper.php	(date 1517481315000)
@@ -274,7 +274,7 @@
 					continue;
 				}
 
-				$streams[] = $file->getBasename('.php');
+				$streams[] = str_replace('stream', '', strtolower($file->getBasename('.php')));
 			}
 		}
 
Index: libraries/classmap.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- libraries/classmap.php	(date 1515624926000)
+++ libraries/classmap.php	(date 1517481181000)
@@ -431,6 +431,6 @@
 JLoader::registerAlias('JStream',                           '\\Joomla\\CMS\\Filesystem\\Stream', '5.0');
 JLoader::registerAlias('JStreamString',                     '\\Joomla\\CMS\\Filesystem\\Streams\\StreamString', '5.0');
 JLoader::registerAlias('JStringController',                 '\\Joomla\\CMS\\Filesystem\\Support\\StringController', '5.0');
-JLoader::registerAlias('JFilesystemWrapperFile',            '\\Joomla\\CMS\\Filesystem\\FileWrapper', '5.0');
-JLoader::registerAlias('JFilesystemWrapperFolder',          '\\Joomla\\CMS\\Filesystem\\FolderWrapper', '5.0');
-JLoader::registerAlias('JFilesystemWrapperPath',            '\\Joomla\\CMS\\Filesystem\\PathWrapper', '5.0');
+JLoader::registerAlias('JFilesystemWrapperFile',            '\\Joomla\\CMS\\Filesystem\\Wrapper\\FileWrapper', '5.0');
+JLoader::registerAlias('JFilesystemWrapperFolder',          '\\Joomla\\CMS\\Filesystem\\Wrapper\\FolderWrapper', '5.0');
+JLoader::registerAlias('JFilesystemWrapperPath',            '\\Joomla\\CMS\\Filesystem\\Wrapper\\PathWrapper', '5.0');
```

**Comment by laoneo** [comment]
Did you mess up the branch?

**Comment by laoneo** [comment]
@wilsonge can we merge this one? Will fix the tests then in a second step.

**Comment by laoneo** [comment]
@wilsonge can we merge this one? Will fix the tests then in a second step.

**Comment by fastslack** [comment]
@laoneo @wilsonge Sorry i tried lot of things to fix this without any result

**Comment by laoneo** [comment]
Removed invalid file with c9f684903b9f7869e2282c3281d623173bfd6274.

**Comment by zero-24** [comment]
@laoneo please use milestones. Thanks 👍

**Comment by laoneo** [comment]
Will do.

**Pull request merged**